### DISCUSSION TITLE
Dimension issues for Swin UNETR implementation

### DISCUSSION THREAD

**Discussion by ghost** [Project-MONAI/tutorials#1695] [Q&A]
Hello, 
I've been trying to implement a Swin UNETR model, using the pretrained weights for BTCV segmentation and the tutorial available on this repository.
However, at the training step, I always run into this error: ValueError: spatial dimensions [2, 3, 4] of input image (spatial shape: torch.Size([73, 73, 34])) must be divisible by 2**5.
The thing is I don't understand why [73 73 34] is the shape of my data, since I used RandCropByPosNegLabeld, with spatial_size = (96, 96, 96)

I tried to resize my data by using Resized in the transforms, but it makes my environment crash...
I then try to resize them later on, in the training process with the line: x = resize_transform(x), with resize_transform = Resized(spatial_dim=(64, 64, 32). This gets me the an error  saying that the dimensions of spatial _dim and the one of my image do not match.
Is there a way to fix this?

**Comment by KumoLiu** [comment]
Hi @bintadkr, could you please paste your transform chain?
And I would suggest you check the data shape after each transform to ensure that you know what shape you get from that transform.

Hope it helps, thanks.

**Reply by ghost** [reply]
Hi @KumoLiu, thank you for your help :)
Here is my transform chain:
num_samples = 4

os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

train_transforms = Compose(
    [
        LoadImaged(keys=["image", "label"]),
        EnsureChannelFirstd(keys=["image", "label"]),
        CropForegroundd(keys=["image", "label"], source_key="image"),
        RandCropByPosNegLabeld(
                keys=["image", "label"],
                label_key="label",
                spatial_size=(96, 96, 96),
                pos=1,
                neg=1,
                num_samples=4,
                image_key="image",
                image_threshold=0,
            ),
        Spacingd(
            keys=["image", "label"],
            pixdim=(1.5, 1.5, 2.0),
            mode=("bilinear", "nearest")
            ),
        EnsureTyped(keys=["image", "label"], device=device, track_meta=True),
    ]
)
output_shape = (1, 407, 407, 68)
val_transforms = Compose(
    [
        LoadImaged(keys=["image", "label"]),
        EnsureChannelFirstd(keys=["image", "label"]),
        #Resized(keys=["label"], spatial_size = [407, 407, 68]),
        CropForegroundd(keys=["image", "label"], source_key="image"),
        Spacingd(
            keys=["image", "label"],
            pixdim=(1.5, 1.5, 2.0),
            mode=("bilinear", "nearest")
            ),
        EnsureTyped(keys=["image", "label"], device=device, track_meta=True),
    ]
)

resize_transforms = Resize(spatial_size = (64, 64, 32))

**Reply by ericspod** [reply]
`CropForegroundd` could be cropping much too tightly for some instances. You can add `SpatialPadd` right after it with `(96,96,96)` for the spatial size to ensure this isn't happening.
